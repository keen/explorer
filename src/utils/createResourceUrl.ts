import { stringify } from './stringify';
import { API_VERSION } from '../constants';

export const createResourceUrl = ({ query, config }) => {
  const { analysis_type: analysisType, ...queryParams } = query;
  const { protocol, host, projectId, masterKey } = config;
  const urlParams = stringify(queryParams);

  return `${protocol}://${host}/${API_VERSION}/projects/${projectId}/queries/${analysisType}?api_key=${masterKey}&${urlParams}`;
};
