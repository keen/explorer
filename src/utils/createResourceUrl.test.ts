/* eslint-disable @typescript-eslint/camelcase */
import { createResourceUrl } from './createResourceUrl';

test('stringify query params', () => {
  const query = {
    filters: [],
    timezone: 'UTC',
    timeframe: 'this_14_weeks',
    zero_fill: null,
    order_by: [
      { direction: 'ASC', property_name: 'result' },
      { direction: 'ASC', property_name: 'platform' },
    ],
    interval: null,
    group_by: ['referrer', 'platform'],
    limit: null,
    event_collection: 'purchases',
  };
  const config = {
    protocol: 'http',
    host: 'host',
    projectId: 'projectId',
    masterKey: 'masterKey',
  };

  expect(createResourceUrl({ query, config })).toMatchSnapshot();
});
