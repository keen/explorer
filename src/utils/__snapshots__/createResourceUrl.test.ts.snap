// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stringify query params 1`] = `"http://host/3.0/projects/projectId/queries/undefined?api_key=masterKey&timezone=0&timeframe=%22this_14_weeks%22&order_by=%5B%7B%22direction%22%3A%22ASC%22%2C%22property_name%22%3A%22result%22%7D%2C%7B%22direction%22%3A%22ASC%22%2C%22property_name%22%3A%22platform%22%7D%5D&group_by=%5B%22referrer%22%2C%22platform%22%5D&event_collection=%22purchases%22"`;
